File: app-mobile/app/src/main/java/com/example/appmobile/models/Projeto.java
package com.example.appmobile.models;

import java.util.List;

public class Projeto {
    private int id;
    private String nome;
    private double orcamentoMaximo;
    private double custoTotalCalculado;
    private int carroId;
    private String criado_em;
    private List<Peca> pecas;

    public Projeto(String nome, double orcamentoMaximo, int carroId, List<Peca> pecas) {
        this.nome = nome;
        this.orcamentoMaximo = orcamentoMaximo;
        this.carroId = carroId;
        this.pecas = pecas;
        this.custoTotalCalculado = pecas.stream().mapToDouble(Peca::getPreco).sum();
    }

    // Getters e Setters
    public int getId() {
        return id;
    }

    public String getNome() {
        return nome;
    }

    public double getOrcamentoMaximo() {
        return orcamentoMaximo;
    }

    public double getCustoTotalCalculado() {
        return custoTotalCalculado;
    }

    public int getCarroId() {
        return carroId;
    }

    public List<Peca> getPecas() {
        return pecas;
    }

    public String getCriado_em() {
        return criado_em;
    }

    @Override
    public String toString() {
        return nome;
    }
}

File: app-mobile/app/src/main/java/com/example/appmobile/models/Projeto.kt
package com.example.appmobile.models

data class Projeto(
    val nome: String,
    val orcamentoMaximo: Double,
    val carroId: Int,
    val pecas: List<Peca>
) {
    val id: Int = 0 // ou var se for inicializado depois
    val custoTotalCalculado: Double = pecas.sumOf { it.preco }
    val criado_em: String? = null // Pode vir da API depois

    override fun toString(): String {
        return nome
    }
}
